### DISCUSSION TITLE
Benefits of lockfile maintenance

### DISCUSSION THREAD

**Discussion by elijaholmos** [renovatebot/renovate#17970] [Suggest an Idea]
Hello, I've done some brief looking around online and especially in Renovate's repo/documentation, and while there's a very quick explanation [here](https://github.com/renovatebot/renovate/issues/1382) about why lockfiles should be maintained, I haven't been able to find something a bit more in-depth that explains some of the pros/cons and why a repo should or should not perform routine lockfile maintenance. I'd like to enable it in some of my repositories, but don't want to blindly turn on the feature simply because "it's good for the codebase". Could a more informed member of the community help point me in the right direction on why this is important or where I can find resources on the topic, especially in a nodejs/pnpm context?

**Comment by rarkins** [comment]
A lock file will essentially "freeze" your dependencies, including transitive, at a point in time. Even if you are regularly upgrading your direct dependencies, npm and Yarn attempt to preserve your other dependencies as much as possible. i.e. don't change them unless they need to be changed. Therefore by default, you may have a lot of outdated dependencies in your lock file. Lock file maintenance is a way of doing a full refresh of the lock file, similar to if you "started over" today and installed from your `package.json` again.
